core/dive.c

```diff
@@ -1328,8 +1328,12 @@ static void fixup_surface_pressure(struct dive *dive)
 			nr++;
 		}
 	}
-	if (nr)
+	if (nr) {
 		dive->surface_pressure.mbar = (sum + nr / 2) / nr;
+		dive->userflags = true;
```

Comment:
so this new `userflags` member puzzles me. It's a plural name, but just a boolean value. And what exactly does it tell me. That the surface pressure is calculated, not set? So should this be `dive->surface_pressure_calculated` ?